solutions/cses-1076.mdx

```diff
@@ -87,3 +82,38 @@ int main() {
 	cout << endl;
 }
 ```
+
+## Solution 2: Order Statistic Tree / Fenwick Tree
+(Arpan Banerjee)
+
+We can directly use an [order statistic tree](https://codeforces.com/blog/entry/11080) (C++) to get the median while sliding the window across the array in $\mathcal{O}(n\log n)$ time.
+
+```cpp
+#include<bits/stdc++.h>
+#include<ext/pb_ds/assoc_container.hpp>
+#include<ext/pb_ds/tree_policy.hpp>
+using namespace std;
+using namespace __gnu_pbds;
+#define ordered_set tree<int, null_type,less_equal<int>, rb_tree_tag,tree_order_statistics_node_update>
+ 
+signed main(){
+        int n, k; cin >> n >> k; int a[n];
+        ordered_set oset;
+        for(int i = 0; i < n; i++){
+                int u; cin >> u; a[i]=u;
+                oset.insert(u);
+                if(i >= k) oset.erase(--oset.lower_bound(a[i-k]));
+                if(i >= k-1) cout << *oset.find_by_order((int)(k / 2.0 - 0.1)) << endl;
```

Comment:
why use floating point numbers?

Reply:
because of the cses definition of median - the median of `1 2 10 20` is `2`

Reply:
```suggestion
                if(i >= k-1) cout << *oset.find_by_order((k - 1) / 2) << endl;
```
@ar492